src/rules/sat_independentset.jl

```diff
@@ -0,0 +1,117 @@
+"""
+The reduction result of a 3-SAT problem to an Independent Set problem.
+"""
+struct Reduction3SATToIndependentSet{T, GT}
+    sat_source::Satisfiability{T}
+    is_target::IndependentSet{GT}
+    k::Int
+    literal_to_nodes::Dict{BoolVar{T}, Vector{Int}}
+end
+target_problem(res::Reduction3SATToIndependentSet) = res.is_target
+
+function reduceto(::Type{<:IndependentSet}, sat_source::Satisfiability)
+    @assert is_kSAT( sat_source.cnf, 3)
+    is_target, k, literal_to_nodes = reduce_3sat_to_independent_set(sat_source)
+    return Reduction3SATToIndependentSet(sat_source, is_target, k, literal_to_nodes )
+end
+
+function extract_solution(res::Reduction3SATToIndependentSet, sol)
+    if length( sol[1] ) >= res.k
+        return transform_is_to_3sat_solution(res.sat_source, sol, res.literal_to_nodes )
+    else
+        return Vector{Int}()
+    end
+end
+
+# ----Useful Functions----
+function reduce_3sat_to_independent_set(s::Satisfiability{T}) where T
```

Comment:
can we generalize this to generic sat?

Reply:
I will have a try! The common references only prove the reduction from 3-SAT to the Independent Set, but the mapping can be more general.